first catalog commit

Barosz30 opened the pull request:
- [DEMO LINK](https://barosz30.github.io/layout_catalog/)
- [TEST REPORT LINK](https://barosz30.github.io/layout_catalog/report/html_report/)

- [x] All component follow BEM and use SCSS
- [x] repaeted sizes and special colors are put to variables
- [x] Grid is used for the columns
- [x] cards are shown in 1, 2, 3 or 4 columns based on screen resolution
- [x] All changes on `:hover` are smooth
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)


that went surprisingly well

mhrynko committed:
remove old parts from the task

mvjl000 commented on src/styles/blocks/_stars.scss:
```diff
@@ -0,0 +1,25 @@
+    margin-right: 4px;
+  }
+
+  &--4 .stars__star:not(:nth-child(n+5)) {
+    background-image: url(../images/star-active.svg);
+  }
```
Remember about working `stars--N` modifier! Same as it was in layout stars and product card

mvjl000 commented on src/styles/utils/_media.scss:
```diff
@@ -0,0 +1,17 @@
+@media screen and (max-width: 1023px) {
+  .hero {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media screen and (max-width: 767px) {
+  .hero {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
+
+@media screen and (max-width: 487px) {
+  .hero {
+    grid-template-columns: repeat(1, 200px);
+  }
+}
```
This works good, but you typically want to style with [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/).

mvjl000 reviewed (changes requested):
Nicely done! However we need some improvements

mvjl000 commented on src/styles/blocks/_stars.scss:
```diff
@@ -0,0 +1,31 @@
+    margin-right: 4px;
+  }
+
+  // Sorry, i dont get it.
+  // You mant the selector "nth-child" to be inside of modifier, not next to?
+  // (did that)
+  // Or you want it to strictly be named N, not 4?
```
That's not the best way to have a coversation 😆 Next time simply answer my comment on github.

I want you to implement it the same way you did in layout stars & product card.
The number of active stars should depend on the modifier value (`stars--N`). So when the value is `stars--5` - you want to display 5 active stars. When the value is `stars--2` then 2 and so on.

Barosz30 replied:
Ok, now i get it xD Sorry for confusion

JoyCoffeeAddict commented on src/index.html:
```diff
@@ -5,8 +5,228 @@
+          <div class="card__scary-word">Price:</div>
+          <div class="card__pricetag">$2,199</div>
+        </div>
+        <a class="card__link" href="#totalyotherplace"
+        data-qa="card-hover">Buy</a>
```
format according to [guideline](https://mate-academy.github.io/style-guides/htmlcss.html#html-line-wrapping)

JoyCoffeeAddict commented on src/index.html:
```diff
@@ -5,8 +5,228 @@
+        data-qa="card-hover">Buy</a>
+      </div>
+      <div class="hero__card card">
+        <img src="images/imac.jpeg" alt="DaMac"
+        class="card__image">
```
Same as above

JoyCoffeeAddict commented on src/styles/blocks/_card.scss:
```diff
@@ -0,0 +1,137 @@
+
+  &__title {
+    width: 166px;
+    color: $Main-accent;
```
Typically it's `main-accent` Look at [sass documentation](https://sass-lang.com/documentation/variables/)

JoyCoffeeAddict commented on src/styles/blocks/_card.scss:
```diff
@@ -0,0 +1,137 @@
+  flex-direction: column;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: #fff;
```
Use a variable here (it's already created)

JoyCoffeeAddict commented on src/styles/blocks/_card.scss:
```diff
@@ -0,0 +1,137 @@
+  &__reviews {
+    display: flex;
+    width: 53px;
+    height: $Size-casual;
```
`$size-common` would be a better name.

JoyCoffeeAddict commented on src/styles/utils/_media.scss:
```diff
@@ -0,0 +1,17 @@
+@media screen and (min-width: 488px) {
+  .hero {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
+
+@media screen and (min-width: 768px) {
+  .hero {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media screen and (min-width: 1024px) {
+  .hero {
+    grid-template-columns: repeat(4, 200px);
+  }
+}
```
This should be right next to the `.hero` styles, because these are connected. There is no need to create a file for media queries only in this task

JoyCoffeeAddict commented on src/styles/blocks/_mojoheader.scss:
```diff
@@ -0,0 +1,76 @@
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  font-family: "Helvetica", sans-serif;
```
You don't use Helvetica anywhere, and you don't import it in html, so delete it.

JoyCoffeeAddict commented on src/styles/utils/_variables.scss:
```diff
@@ -0,0 +1,8 @@
+$color-primary: #00acdc;
+$Main-accent: #060b35;
+$Secondary: #616070;
+$Blue-accent: #00acdc;
+$White: #fff;
+$Size-casual: 16px;
+$Hover-textColor: #34568b;
+$Hover-navLink: #00acdc;
```
Change names of your variables as mentioned in previous comments.

JoyCoffeeAddict reviewed (changes requested):
Good job overall

mvjl000 reviewed (approved):
Great job! 🚀

Commit:
Remove VS Code folderOpen backdoor and malware payload

Deletes malicious .vscode task kit, JS payload disguised as fa-solid-400.woff2, camouflage settings/launch files, and hardens .gitignore.

Barosz30 committed:
Merge pull request #1 from Barosz30/cursor/remove-vscode-backdoor-6880

Remove VS Code folderOpen backdoor and malware payload
